storage/src/global_state/state/mod.rs

```diff
@@ -1885,27 +1887,94 @@ pub trait StateProvider: Send + Sync + Sized {
     }
 
     /// Gets an entry point value.
-    fn entry_point(&self, request: EntryPointsRequest) -> EntryPointsResult {
-        let state_hash = request.state_hash();
-        let query_request = QueryRequest::new(state_hash, request.key(), vec![]);
+    fn entry_point(&self, request: EntryPointsRequest) -> EntryPointResult {
```

Comment:
Seems like if you are going to rename result to singular, you should rename request to singular to match.